src/HappyConnOpener.cc

```diff
@@ -796,3 +812,13 @@ HappyConnOpener::maybeOpenSpareConnection()
     return false;
 }
 
+/// Check for maximum connection tries and forwarding time restrictions
+bool
+HappyConnOpener::ranOutOfTimeOrAttempts() const
+{
+    if (n_tries >= maxTries)
+        return true;
+
+    return (FwdState::ForwardTimeout(fwdStart) == 0);
```

Comment:
This "works" because ForwardTimeout() is careful, but please use a less scary `<=` instead of `==` to reduce the chances that the code reader will decide she needs to double check ForwardTimeout() implementation.

Reply:
resolved at 05190e3